File: leetcode2/0590. N-ary Tree Postorder Traversal.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution(object):
    def postorder(self, root):
        """
        :type root: Node
        :rtype: List[int]
        """
                
        if not root:
            return []
        root_list=[root]
        rep=[]

        while root_list:
            cur_root=root_list.pop()
            rep.append(cur_root.val)
            if cur_root.children:
                for child in cur_root.children:
                    root_list.append(child)
        return rep[::-1]

File: C#/0590. N-ary Tree Postorder Traversal.cs
/*
// Definition for a Node.
public class Node {
    public int val;
    public IList<Node> children;

    public Node(){}
    public Node(int _val,IList<Node> _children) {
        val = _val;
        children = _children;
}
*/
public class Solution {
    public IList<int> Postorder(Node root) {
        IList<int> rep=new List<int>();
        if(root==null){
            return rep;
        }
        Stack<Node> rootList=new Stack<Node>();
        rootList.Push(root);
        
        while (rootList.Count!=0){
            Node curRoot=rootList.Pop();
            rep.Insert(0,curRoot.val);
            foreach(Node child in curRoot.children){
                rootList.Push(child);
            }
        }
        return rep;
    }
}
